### PR TITLE
Add Box Seeding Documentation to Segmentation Parameter Docs

### LINKED ISSUE #399: Add "box" seeding to segmentation documentation
At present, the [documentation for segmentation](https://tobac.readthedocs.io/en/latest/segmentation.html) does not mention the box seeding option. As this is both an important addition as part of the v1.5 release, and probably the best practice for 3D segmentation we should aim to provide some documentation and possibly a comparison between box and column seeding

### PR DESCRIPTION
**Pull request by freemansw1**
Resolves #399 and adds box seeding to our documentation for segmentation parameters.

Worked on with @RBhupi during the tobathon 2024

* [x] Have you followed our guidelines in CONTRIBUTING.md? 
* [x] Have you self-reviewed your code and corrected any misspellings? 
* [x] Have you written documentation that is easy to understand?
* [x] Have you written descriptive commit messages? 
* [ ] Have you added NumPy docstrings for newly added functions? 
* [ ] Have you formatted your code using black? 
* [ ] If you have introduced a new functionality, have you added adequate unit tests?
* [ ] Have all tests passed in your local clone? 
* [x] If you have introduced a new functionality, have you added an example notebook?
* [x] Have you kept your pull request small and limited so that it is easy to review? 
* [x] Have the newest changes from this branch been merged?

### CODE REVIEW

**Comment by freemansw1** [comment]
@harrietgilmour and @w-k-jones, are you available to review this PR within the next week or so?

**Comment by harrietgilmour** [comment]
> @harrietgilmour and @w-k-jones, are you available to review this PR within the next week or so?

Yes no problem. I can do it next week

**Comment by freemansw1** [comment]
> > @harrietgilmour and @w-k-jones, are you available to review this PR within the next week or so?
> 
> Yes no problem. I can do it next week

Thanks! Trying to be more deliberate with our review requests.

File: doc/segmentation_parameters.rst [hunk 1]
```diff
@@ -8,35 +8,62 @@ A full list of parameters and descriptions can be found in the API Reference: :p
...
+
+**Column** seeding (:code:`seed_3D_flag='column'`) works by setting seed markers throughout some or all of the vertical column at all detected feature centroids (i.e., one column per feature detected). While the default behavior is to seed throughout the full vertical column, the vertical extent of the seeds can be set by passing a `slice <https://docs.python.org/3/c-api/slice.html>`_ into the :code:`level` parameter. Note that :code:`level` operates in *array* coordinates rather than physical coordinates.
+
+**Box** seeding (:code:`seed_3D_flag='box'`) sets a cube or rectangular seed markers around the detected feature in 3D space. The marker size is user defined (in array coordinates) by :code:`seed_3D_size` as either an integer (for a cube) or a tuple of :code:`(int, int, int)`, ordered :code:`(vertical, hdim_1, hdim_2)`. Note that :code:`seed_3D_size` must be odd number to avoid the box becoming biased to one side. If two seed boxes overlap, the seeded area is marked with the closest feature centroid.
```
Comment (harrietgilmour) [inline doc/segmentation_parameters.rst:59]: missing 'an' from "Note that ... must be {an} odd number to avoid the box becoming biased to one side"

**Review by harrietgilmour — APPROVED** [review]
All looks good - clear and concise. Just added a comment to add in a missing "an" but this is very minor.

**Comment by w-k-jones** [comment]
Also the docstring are wrongly formatted for `seed_3D_flag` in `segmentation.py` which is causing the API reference to render incorrectly. I can fix that quickly and add it to this PR

File: doc/segmentation_parameters.rst (hunk 1 above)
Comment (freemansw1) [reply doc/segmentation_parameters.rst:59]: Good catch, thanks!

**Comment by freemansw1** [comment]
Thanks @w-k-jones @harrietgilmour ! 

@w-k-jones what is our policy on documentation-only PRs? We can merge these right into `main`, right?

**Comment by w-k-jones** [comment]
> Thanks @w-k-jones @harrietgilmour !
> 
> @w-k-jones what is our policy on documentation-only PRs? We can merge these right into `main`, right?

Yes, looks like there's no issues with merging into main so I'm happy for you to do that directly

**Pull request merged**